tests/test_dataelem.py

```diff
@@ -1251,3 +1261,199 @@ def test_pixel_data_ndarray_raises(self):
             ds.PixelData = np.ones((3, 4), dtype="u1")
 
         assert ds.PixelData == b"\x00\x01"
+
+
+class TestBufferedDataElement:
+    """Tests setting a DataElement value to a buffer"""
+
+    @pytest.mark.parametrize("vr", BUFFERABLE_VRS)
+    def test_reading_dataelement_buffer(self, vr):
+        value = b"\x00\x01\x02\x03"
+        buffer = io.BytesIO(value)
+        elem = DataElement("PixelData", vr, buffer)
+
+        data: bytes = b""
+        # while read_bytes is tested in test_buffer.py, this tests the integration
+        # between the helper and DataElement since this is the main use case
+        for chunk in read_buffer(elem.value):
+            data += chunk
+
+        assert data == value
+
+    def test_unsupported_vr_raises(self):
+        """Test using a buffer with an unsupported VR raises"""
+        msg = (
+            "Elements with a VR of 'PN' cannot be used with buffered values, "
+            "supported VRs are: OB, OD, OF, OL, OV, OW"
+        )
+        with pytest.raises(ValueError, match=msg):
+            DataElement("PersonName", "PN", io.BytesIO())
+
+    @pytest.mark.skipif(IS_WINDOWS, reason="Running on Windows")
```

Comment:
Maybe clarify that it's because of `TemporaryFile` not working as buffer under Windows.

Reply:
Done, technically it's because it doesn't have write-only mode.